signing_service/signing_service/signing_service.py

```diff
@@ -106,9 +127,28 @@ def _connect(self, tcp_socket: socket.socket) -> None:
 
     def _handle_connection(self, tcp_socket: socket.socket) -> None:  # pylint: disable=no-self-use
         """ Inner loop that handles data exchange over socket. """
-        while True:
-            data = tcp_socket.recv(1024)
-            tcp_socket.send(data)
+        receive_frame_generator = unescape_stream(connection=tcp_socket)
+
+        for raw_message_received in receive_frame_generator:
+            try:
+                AbstractFrame.deserialize(
+                    raw_message=raw_message_received,
+                    public_key=self.signing_service_public_key,
+                )
+            except (InvalidSignature, MiddlemanProtocolError) as exception:
+                logger.info(f'Malformed messages failed to deserialize with exception: {exception}.')
+                middleman_message_response = ErrorFrame(
+                    payload=(ErrorCode.InvalidFrameSignature, str(exception)),
+                    request_id=99,
```

Comment:
I would suggest making it `0` (restricted `request_id`) and extracting to a (protocol) constant

Reply:
:+1: done